WordPress/Classes/ViewRelated/Post/PostListViewController.swift

```diff
@@ -0,0 +1,685 @@
+            // Don't allow editing while pushing changes
+            return
+        }
+        
+        if post.status == PostStatusTrash {
+            // No editing posts that are trashed.
+            return
+        }
+        
+        previewEditPost(post)
+    }
+    
+    private func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
+        let post = tableViewHandler?.resultsController.objectAtIndexPath(indexPath) as! Post
+        
+        let identifier = cellIdentifierForPost(post)
+        let cell = tableView.dequeueReusableCellWithIdentifier(identifier, forIndexPath: indexPath)
+        
+        configureCell(cell, atIndexPath: indexPath)
+        
+        return cell
+    }
+    
+    private func configureCell(cell: UITableViewCell, atIndexPath indexPath: NSIndexPath) {
+        cell.accessoryType = .None
+        cell.selectionStyle = .None
+        
+        if let postCell = cell as? PostCardCell {
+            postCell.delegate = self
+            
+            if let post = tableViewHandler?.resultsController.objectAtIndexPath(indexPath) as? WPPostContentViewProvider {
+                let layoutOnly = (cell == imageCellForLayout) || (cell == textCellForLayout)
+                
+                postCell.configureCell?(post, layoutOnly: layoutOnly)
+            }
+        }
+    }
+    
+    private func cellIdentifierForPost(post: Post) -> String {
+        var identifier : String
+        
+        if recentlyTrashedPostObjectIDs?.containsObject(post.objectID) == true && currentPostListFilter()?.filterType != .Trashed {
+            identifier = self.dynamicType.postCardRestoreCellIdentifier
+        } else if post.pathForDisplayImage?.characters.count > 0 {
+            identifier = self.dynamicType.postCardTextCellIdentifier
+        } else {
+            identifier = self.dynamicType.postCardImageCellIdentifier
+        }
+        
+        return identifier
+    }
+    
+    // MARK: - Post Actions
+    
+    private func createPost() {
+        if WPPostViewController.isNewEditorEnabled() {
+            createPostInNewEditor()
+        } else {
+            createPostInOldEditor()
+        }
+    }
+    
+    private func createPostInNewEditor() {
+        let postViewController = WPPostViewController(draftForBlog: blog)
+        
+        postViewController.onClose = { [weak self] (viewController, changesSaved) -> () in
+            if changesSaved {
+                self?.setFilterWithPostStatus(viewController.post.status)
+            }
+            
+            viewController.presentingViewController?.dismissViewControllerAnimated(true, completion: nil)
+        }
+        
+        let navController = UINavigationController(rootViewController: postViewController)
+        navController.restorationIdentifier = WPEditorNavigationRestorationID
+        navController.restorationClass = WPPostViewController.self
+        navController.toolbarHidden = false
```

Comment:
It would be nice to keep the annotation from the original objc source as to why this is being set.